Sources/CvPlot.cpp

```diff
@@ -15803,39 +15373,26 @@ CvUnit* CvPlot::getWorstDefender(PlayerTypes eOwner, PlayerTypes eAttackingPlaye
 
 bool CvPlot::isInUnitZoneOfControl(PlayerTypes ePlayer) const
 {
-	for (int iI = 0; iI < NUM_DIRECTION_TYPES; ++iI)
+	foreach_(CvPlot* adjacentPlot, adjacent())
 	{
-		const CvPlot* pAdjacentPlot = plotDirection(getX(), getY(), (DirectionTypes)iI);
-		if (pAdjacentPlot != NULL)
+		foreach_(CvUnit* unit, adjacentPlot->units() | filtered(CvUnit::fn::isZoneOfControl()))
```

Comment:
This is technically correct but doesn't make it clear what the intent is.
Better would be:
```cpp
bst::optional<CvUnit*> zocUnit = algo::find_if(adjacentPlot->units(), CvUnit::fn::isZoneOfControl());
if(zocUnit)
{
	return GET_TEAM(zocUnit->getTeam()).isAtWar(GET_PLAYER(ePlayer).getTeam());
}
```
However I think the logic is wrong anyway? Surely it should look if *any* zoc units are at war with the player, not just the first one it finds?
Like:
```cpp
if(algo::any_of(adjacentPlot->units(), CvUnit::fn::isZoneOfControl() && CvUnit::fn::isAtWar(ePlayer))
{
	return true;
}
```
(need to add the isAtWar function for units for that one).